File: redsheet-portal/src/app/database-models/their-team-member.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { Personality, ProjectNegotiation,  vTheirTeamMemberRecord  } from './'
export class TheirTeamMember {
    constructor(theirTeamMember?: TheirTeamMember  | vTheirTeamMemberRecord )  {
        let properties = [ "position", "theirTeamMemberId", "theirTeamMemberName", "personalityId", "projectNegotiationId",];
        for (let property of properties)
            if (theirTeamMember && theirTeamMember[property])
                this[property] = theirTeamMember[property];
    }

    @required()
    @maxLength(50)
	position : string =   undefined;

	theirTeamMemberId : number =   0 ;

    @required()
    @maxLength(50)
	theirTeamMemberName : string =   undefined;

    @range(0,2147483647)
	personalityId : number =   undefined;
	personality : Personality  ;

    @range(0,2147483647)
	projectNegotiationId : number =   undefined;
	projectNegotiation : ProjectNegotiation  ;


}

File: RedSheet.DbEntities/Models/TheirTeamMember.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("TheirTeamMembers", Schema = "dbo")]
	[Serializable]
	public partial class TheirTeamMember  
	{
 
	[Required] 
	[MaxLength(500)]
	public string Position { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int TheirTeamMemberId { get; set; } 
 
	[Required] 
	[MaxLength(50)]
	public string TheirTeamMemberName { get; set; } 
 
	[RelationshipTableAttribue("Personalities","dbo")]
	//Relationships
	public int PersonalityId { get; set; } 
 
	[ForeignKey("PersonalityId")]
	public Personality Personality { get; set; } 
 
	[RelationshipTableAttribue("ProjectNegotiations","dbo")]
	//Relationships
	public int ProjectNegotiationId { get; set; } 
 
	[ForeignKey("ProjectNegotiationId")]
	public ProjectNegotiation ProjectNegotiation { get; set; } 

	public TheirTeamMember()
	{
    }

    }
}

